packages/block-editor/src/hooks/align.js

```diff
@@ -207,6 +212,130 @@ export const withDataAlign = createHigherOrderComponent(
 		return <BlockListBlock { ...props } wrapperProps={ wrapperProps } />;
 	}
 );
+export function AlignmentVisualizer( {
+	value = 'none',
+	allowedAlignments,
+	clientId,
+} ) {
+	const layout = useLayout();
+	const blockName = useSelect(
+		( select ) => select( blockEditorStore ).getBlockName( clientId ),
+		[ clientId ]
+	);
+
+	const [ editorWidth, setEditorWidth ] = useState( 0 );
+	const ref = useRefEffect( ( node ) => {
+		const { ownerDocument } = node;
+		const { defaultView } = ownerDocument;
+		const editorWrapperElement = ownerDocument.querySelector(
+			'.editor-styles-wrapper'
+		);
+		const updateWidth = () =>
+			setEditorWidth( editorWrapperElement.offsetWidth );
+		updateWidth();
+
+		const resizeObserver = defaultView.ResizeObserver
+			? new defaultView.ResizeObserver( updateWidth )
+			: undefined;
+		resizeObserver?.observe( editorWrapperElement );
+
+		return () => {
+			resizeObserver?.disconnect();
+		};
+	}, [] );
+
+	const blockAllowedAlignments = getValidAlignments(
+		getBlockSupport( blockName, 'align' ),
+		hasBlockSupport( blockName, 'alignWide', true )
+	);
+
+	// Allow override of `blockAllowedAlignments`. The image block doesn't use
+	// alignment block supports, so this allows the image block to directly
+	// declare what it supports.
+	const availableAlignments = useAvailableAlignments(
+		allowedAlignments ?? blockAllowedAlignments
+	);
+
+	const alignments = useMemo( () => {
+		return [ 'none', 'wide', 'full' ]
```

Comment:
```suggestion
		return availableAlignments
```

I appreciate this is still under development but should this be using the `availableAlignments`?

One of the oddities I spotted while testing was due to an alignment that didn't exist for the theme still being included. This made the wide and full alignments overlap regardless of size for that theme.

Also, `editorWidth` doesn't appear to be used within this `useMemo`.

Reply:
Yep, I didn't mean to leave that in 😄